enhancements/installer/assisted-installer-in-cluster.md

```diff
@@ -0,0 +1,539 @@
+expect before beginning installation for each of the control-plane and worker
+roles.
+
+#### Use of metal3
+
+[metal3](https://metal3.io/) will be used to boot the assisted discovery ISO
+for bare-metal deployments. Specifically, the BareMetalHost API will be used
+from a Hub cluster to boot the discovery ISO on hosts that will be used to
+form new clusters.
+
+For this first iteration, it is assumed that the hub cluster itself is using
+the bare metal platform, and thus will have baremetal-operator installed
+and available for use. In the future it may be desirable to make the metal3
+capabilities available on hub clusters that are not using the bare metal
+platform.
+
+#### BareMetalHost can boot live ISOs
+
+A separate [enhancement to
+metal3](https://github.com/metal3-io/metal3-docs/pull/150) proposes a new
+capability in the BareMetalHost API enabling it to boot live ISOs other than
+the Ironic Python Agent. That feature is required so that automation in a
+cluster can boot the discovery ISO on known hardware as the first step toward
+provisioning that hardware.
+
+#### Host Approval
+
+It is important that when an agent checks in, it not be allowed to join a
+cluster until an actor has approved it. From a security standpoint, it is not
+OK for anyone who can access a copy of a discovery ISO and/or access the API
+where agents report themselves to implicitly have the capability to join a
+cluster as a Node.
+
+The Agent CRD will have a field in which to designate that it is approved. If
+the host was booted via the baremetal-operator, approval will be granted
+automatically.
+
+#### Day 2 Add Node Boot-it-Yourself
+
+#### Personas for multi-cluster management
+
+**Infra Owner** Manages physical infrastructure and configures hosts to boot
+the discovery ISO that runs the Agent.
+
+**Cluster Creator** Uses running Agents to create and grow clusters.
+
+#### Create Cluster
+
+This scenario takes place on a hub cluster where hive and possibly RHACM are
+present. This scenario does not include centralized machine management.
+
+1. Infra Owner creates an InstallEnv resource. It can include fields such as egress proxy, NTP server, SSH public key, ... In particular it includes an Agent label selector, and a separate field of labels that should be applied to Agents.
+1. Infra Owner creates BareMetalHost resources that include BMC credentials. They are labeled so that they match the selctor on the InstallEnv.
+1. A new controller, the Baremetal Agent Controller, sees the matching BareMetalHosts and boots them using the discovery ISO URL found in the InstallEnv's status.
+1. The Agent starts up on each host and reports back to the assisted service, which creates an Agent resource in the cluster. The Agent is labeled with the labels that were specified in the InstallEnv's Spec.
+1. The Baremetal Agent Controller sets the Agent's Role field in its spec to "master" or "worker" if a corresponding label was present on its BareMetalHost.
```

Comment:
Those are the most common roles, but not the only ones. For example, we've had discussions on the mailing list recently about things like remote workers with special roles.

Why is the role coming from the host? Why isn't setting the role part of approving the agent in the first place?

Reply:
> Those are the most common roles, but not the only ones. For example, we've had discussions on the mailing list recently about things like remote workers with special roles.

Right, but I don't think other roles are supported during cluster creation.

> 
> Why is the role coming from the host? Why isn't setting the role part of approving the agent in the first place?

This is for users who want to describe everything up-front and let automation turn BareMetalHosts into clusters, for example in a gitops approach. The role does not have to be specified that way, but it is the way a user could specify it and get an end-to-end automated install. I'll add a note about this before the enumerated steps.